src/zarr/array_spec.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Literal
+
+import numpy as np
+
+from zarr.buffer import Prototype
+from zarr.common import ChunkCoords, parse_dtype, parse_fill_value, parse_order, parse_shapelike
+
+
+@dataclass(frozen=True)
+class ArraySpec:
+    shape: ChunkCoords
+    dtype: np.dtype[Any]
+    fill_value: Any
+    order: Literal["C", "F"]
+    prototype: Prototype
```

Comment:
Is the `Prototype` pickleable?

Reply:
The `default_prototype`, which is the one we use in [`ArrayV3Metadata`](https://github.com/zarr-developers/zarr-python/pull/1910/files#diff-6ad3ea3fee31fe35b9e5c70976ad07c8ade9674b1e60578af98be6ab1cdd7ec2R199),  is pickleable. 
AFAIK, it is not possible to type hint _pickable_ but added the requirement in the docstring.

Reply:
The thing that has to be pickleable is the class, not the object itself. I'm having a hard time imagining any scenario in which a class is not pickleable.